File: C/Algorithms/2389.c
#include "math.h"

int cmpint(const void *x, const void *y) {
    int a = *(const int *)x;
    int b = *(const int *)y;
    
    return (a > b) - (a < b);
}

/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* answerQueries(int* nums, int numsSize, int* queries, int queriesSize, int* returnSize){
    qsort(nums, numsSize, sizeof(*nums), cmpint);
    
    int acc = 0;
    for(int i = 0; i < numsSize; i++) {
        nums[i] += acc;
        acc = nums[i];
    }
    
    int *result = malloc(queriesSize * sizeof(int));
    for (int i = 0; i < queriesSize; i++) {
        int needle = queries[i], index = 0;
        // Binary Search:
        int left = 0, right = numsSize;
        while (left < right) {
            int m = (left + right) / 2;
            if (nums[m] > needle) 
                right = m;
            else 
                left = m + 1;
        }
        result[i] = right;
    }
    *returnSize = queriesSize;
    return result;
}

File: Rust/Algorithms/2389.rs
impl Solution {
    pub fn answer_queries(mut nums: Vec<i32>, queries: Vec<i32>) -> Vec<i32> {
        nums.sort_unstable();
        
        let mut acc = 0;
        for i in 0..nums.len() {
            nums[i] += acc;
            acc = nums[i];
        }
        
        let mut result = vec![0; queries.len()];
        for i in 0..queries.len() {
            result[i] = nums
                .binary_search(&queries[i])
                .map_or_else(|e| e, |o| o + 1) as i32;
        }
        result
    }
}